### ISSUE TITLE
Improves Admin pages editor

### ISSUE THREAD

**Issue by decabeza** [consul/consul#2495]
Background
==

Now in the Admin Panel, there is a section to create pages `/admin/site_customization/pages` and we're having problems with the content that it generates.

![ck_editor](https://user-images.githubusercontent.com/631897/36801742-2ddd81d2-1cb3-11e8-96f3-f31542a313e5.png)

In `/views/pages/custom_page.html.erb` the content is rendered with 
`<%= text_with_links @custom_page.content %>` but doesn't allow the user to insert headings or links.

What we need
==

Replace **CK Editor** with another tool or update it to [CK Editor 5](https://github.com/ckeditor/ckeditor5) to change the configuration and allow:
- Insert headings
- Insert links
- Optional but wonderful: Insert images

The resulting content should have a correct HTML structure and format.

Screenshot
==
### **`CK Editor 5`**
![ckeditor5](https://user-images.githubusercontent.com/631897/36802157-739d0822-1cb4-11e8-8b24-15f474b2a4d0.png)

Other option to explore
==

[Trix editor](https://trix-editor.org/) used on basecamp. [Trix repo](https://github.com/basecamp/trix)

**Comment by raul-fuentes** [comment]
I'll do this one

**Comment by raul-fuentes** [comment]
Hi @decabeza, this is the actual look of the field with ckeditor5 (with a very basic setup)
![screenshot from 22](https://user-images.githubusercontent.com/33748390/40615529-6d027d76-6287-11e8-8c97-4f334ed0377b.png)


Before starting with the images I must warn that it will be a bit long, you have to add an image uploader to each model where the editor is used.
I imagine that the imageable concern can handle it but I would like to have the opinion of  @voodoorai2000 

Apart from that, should I include more things in the editor's toolbar?

**Comment by voodoorai2000** [comment]
Sure @raul-fuentes the current imageable concern should work fine integrating it with ckeditor 👍 

Thinking about the upcoming [migration from paperclip to ActiveStorage](https://github.com/consul/consul/issues/2629) it's probably helpful to send the image upload for ckeditor in a separate PR 👌

**Comment by decabeza** [comment]
@raul-fuentes I think this basic setup is enough! 👏  And how @voodoorai2000 says, better send the image upload option in a separate PR 😉

**Comment by javierm** [comment]
Closed via #2876.
